### ISSUE TITLE
You need to update your app to join this server

### ISSUE THREAD

**Issue by dmulrooney** [V4NSH4J/discord-mass-DM-GO#396]
What is this error? I was able to solve captchas using the anti-captcha plugin, but I get this error when posting captcha solution and tq token.

I would like to work with the community in fixing this.

**Comment by V4NSH4J** [comment]
Hey! Thank you for this issue. For me, this is almost always the response when a captcha pops up. I wouldn't worry about it, it doesn't affect much. I'll try out the anti-captcha.com plugin as well, great idea

**Comment by dmulrooney** [comment]
V4NSH4J, glad to hear from you.
I got it to the point where it appears to be working from a captcha point of view. However, once I solve the captcha, even manually, it is just reloading the invite page and making me solve it again. It might just be my tokens/proxies at this point though.

Errors I've seen are "you need to update your app to join this server" .

**Comment by dmulrooney** [comment]
Example of server captcha solve using anticaptcha plugin:

captcha_key: "P0_eyJ0eXAi...wLJhg"
captcha_rqtoken: "IjVObmk1UC94WDhJY...Iagm8LVlFM9Z-GLg"

returns

**{"captcha_key": ["captcha-required"]**, "captcha_sitekey": "a9b5fb07-92ff-493f-86fe-352a2803b3df", "captcha_service": "hcaptcha", "captcha_rqdata": "oXdn4.....1sbLCf", "captcha_rqtoken": "InJE......GN_3Q"}

**Comment by V4NSH4J** [comment]
Same thing happened to me too a few days ago
https://www.youtube.com/watch?v=Xk7cP9hotnM&feature=youtu.be

I don't think anticaptcha supports these new sitekeys. Only services I know of that do are 2captcha and rucaptcha

**Comment by dmulrooney** [comment]
Just writing this for documentation purposes.
1. If we submit the captcha as normal, sometimes it works. If so, let it work.
2. Sometimes, the captcha fails with this error:
    {'captcha_key': ['You need to update your app to join this server.'], 'captcha_sitekey': 'a9b5fb07-92ff-493f-86fe-352a2803b3df', 'captcha_service': 'hcaptcha', 'captcha_rqdata': 'C8Na/v+oCXxvlKTVRXZZAMgMFY2ahd1FRXfchPCXLE0apVAAsZe5tArK3eU8PeGG4JROubem+/NjWB1Tg8qFgCi5mFgvjLRV41JyY+A60nTD2329Y8blEEcepoDJXaftFLJpYSJHLKG1XOSwABtRyXwgaHtBAO0Z2zZ20n5W', 'captcha_rqtoken': 'IlNWc1FldVVORWQzbHBQSHUwT3JhbnFpV2NBdnJaNi9VbnozVlVwS2lZYzA4WllNKzU5ajFVNkF5bGo3Ly9PckdrVGhIR0E9PTV2bklqRHA5QUJZMktmRVAi.YikleA.jfic4im23JgCpxUxco98uoDiOHE'}
3. When this happens, take the "captcha_sitekey" and "captcha_rqdata".
4. Solve the server join captcha from scratch (new solve), using the response sitekey, and response rqtoken/rqdata.
5. Profit

I am implementing this now and will let you know if it works. Only thing I'm not 100% sure on is that I need to redo a fresh captcha, I think so.

**Comment by V4NSH4J** [comment]
Yes, the infinite captcha loop happens on some tokens manually as well. You can submit the RqData to 2captcha in a data parameter and also specify the Useragent in userAgent parameter.

**Comment by dmulrooney** [comment]
1. Where do I get the RqData, in the first captcha solve's response? Do I just resubmit a second captcha if so with rqdata from the first response?
3. Does dmdgo not specify useragent currently?

**Comment by V4NSH4J** [comment]
> 1. Where do I get the RqData, in the first captcha solve's response? Do I just resubmit a second captcha if so with rqdata from the first response?
> 
>     2. Does dmdgo not specify useragent currently?

DMDGO specifies the userAgent everywhere it can.
You get the RqData along with the hcaptcha sitekey in the 400 bad request's response body from Discord when joining server

**Comment by Gercekefsane** [comment]
There is no solution to this with the last update. It asks for Captcha in every request made with Discord request. If you solve the captcha, discord will ban you.

**Comment by texascrypto** [comment]
I really want to solve this guys. What can I try?

**Comment by V4NSH4J** [comment]
#488
